src/preact/base-element.js

```diff
@@ -513,7 +513,10 @@ export class PreactBaseElement extends AMP.BaseElement {
           this.hydrationPending_ = true;
         } else {
           // Create new shadow root.
-          shadowRoot = this.element.attachShadow({mode: 'open'});
+          shadowRoot = this.element.attachShadow({
+            mode: 'open',
+            delegatesFocus: true,
```

Comment:
See [this thread](https://github.com/ampproject/amphtml/pull/31994#discussion_r560496531). This makes a lot of sense, but not for all components. Let's make it configurable via the `PreactBaseElement['delegatesFocus']`, make it `false` by default, and let's also test it.

Reply:
Great, that makes sense.  Will follow up with a new revision.